File: CPP-Algorithms/Selection_Sort.cpp
#include <iostream>

void selectionSort(int arr[], int n) {
    for (int i = 0; i < n - 1; i++) {
        int minIndex = i;
        for (int j = i + 1; j < n; j++) {
            if (arr[j] < arr[minIndex]) {
                minIndex = j;
            }
        }
        std::swap(arr[i], arr[minIndex]);
    }
}

int main() {
    int n;
    std::cout << "Enter the number of elements in the array: ";
    std::cin >> n;

    int arr[n];
    for (int i = 0; i < n; i++) {
        std::cout << "Enter the elements of the array "<<i<<" : ";
        std::cin >> arr[i];
    }

    selectionSort(arr, n);

    std::cout << "Sorted array: ";
    for (int i = 0; i < n; i++) {
        std::cout << arr[i] << " ";
    }
    std::cout << std::endl;

    return 0;
}

File: Hacktoberfest-2023/Programming/Kotlin/SelectionSort.kt
fun selectionSort(arr: IntArray) {
  val n = arr.size
  for (i in 0 until n - 1) {
    var minIndex = i
    for (j in i + 1 until n) {
      if (arr[j] < arr[minIndex]) {
        minIndex = j
      }
    }
    // Swap arr[i] and arr[minIndex]
    val temp = arr[i]
    arr[i] = arr[minIndex]
    arr[minIndex] = temp
  }
}
fun main() {
  val arr = intArrayOf(64, 25, 12, 22, 11)
  println("Before sorting: ${arr.joinToString(", ")}")

  selectionSort(arr)

  println("After sorting: ${arr.joinToString(", ")}")
}